Drv/Ip/TcpClientSocket.cpp

```diff
@@ -57,12 +57,22 @@ SocketIpStatus TcpClientSocket::openProtocol(NATIVE_INT_TYPE& fd) {
 #if defined TGT_OS_TYPE_VXWORKS || TGT_OS_TYPE_DARWIN
     address.sin_len = static_cast<U8>(sizeof(struct sockaddr_in));
 #endif
+    if (reuse_address)
```

Comment:
Based on the comments, I didn't expect this code in the client....only in the server.  If that is true, then perhaps it is only needed in the `startup` function and not the `open` functions?

Reply:
I just tried to run the unit tests with the 4 different configurations, and it seems like both sockets are needing the REUSEADDR option to avoid errors. You can double check from your side using the procedure mentioned in the issue https://github.com/nasa/fprime/issues/2706 on the TcpClient UT.

To be honest, I am not an expert in TCP sockets, and I can not really explain the real reason behind this behavior, sadly.

Reply:
I can attempt to explain why these errors occur.  TCP has two endpoints (a sever and a client).  When the connection is terminated there are two termination possibilities:

1. Uncoordinated termination (typically caused by calling `close` before calling `shutdown`)
2. Coordinated termination (caused by calling `shutdown` and then `close`)

In the uncoordinated case, a port will end-up in `TIME_WAIT*` state, where it waits for a specified timeout (typically 60s) to let any remote that was not properly informed to shutdown notice the connection is down before allowing the port to be reused.  Otherwise, the client might send data to a reused port accidentally leaking data.

In the coordinated case, a message is sent across TCP informing the remote to shutdown and an acknowledgement is sent back.  Then the port can be closed, and immediately reused.

These errors occur when the port cannot be reused immediately due to an uncoordinated shutdown, or an unacknowledged coordinated shutdown.  Thus the port is unable to be reused, but the UTs expect it can.

Given the random nature of these errors, it is hard to reproduce with enough insight to determine which of these cases occur and exactly why.  In an ideal world, the UTs would conduct a proper `shutdown` always, and thus not need the ability to "reuse" waiting ports.  However, your proposed solution is deeply practical as it quickly resolves the issue without trying to reproduce a random bug.

Reply:
@LeStarch Thanks for the explanations.

What would be the path forward for this MR? 
Do you want me to investigate more on the proper shutdown of the sockets for the UT instead of having the reuse option?

Reply:
I like the reuse option as a fallback. Especially given the time required to figure out the cause of the other.

Reply:
Do you want any other changes in the current MR then?

I will anyways investigate on the proper shutdown of the UT in the future, and come back to you if I have any findings.